fsspec/dircache.py

```diff
@@ -45,36 +53,36 @@ def __init__(
             The number of most recent listings that are considered valid; 'recent'
             refers to when the entry was set.
         """
-        self._cache = {}
-        self._times = {}
+        self._cache: dict[str, list[DirEntry]] = {}
+        self._times: dict[str, float] = {}
         if max_paths:
             self._q = lru_cache(max_paths + 1)(lambda key: self._cache.pop(key, None))
         self.use_listings_cache = use_listings_cache
         self.listings_expiry_time = listings_expiry_time
         self.max_paths = max_paths
 
-    def __getitem__(self, item):
+    def __getitem__(self, item: str) -> list[DirEntry]:
         if self.listings_expiry_time is not None:
             if self._times.get(item, 0) - time.time() < -self.listings_expiry_time:
                 del self._cache[item]
         if self.max_paths:
             self._q(item)
         return self._cache[item]  # maybe raises KeyError
 
-    def clear(self):
+    def clear(self) -> None:
         self._cache.clear()
 
-    def __len__(self):
+    def __len__(self) -> int:
         return len(self._cache)
 
-    def __contains__(self, item):
+    def __contains__(self, item: object) -> bool:
         try:
-            self[item]
+            self[item]  # type: ignore[index]
```

Comment:
This is a little unfortunate -- the `MutableMappign` typehints say that `__contrains__` should take all `object`s, not just the key type, hence the type ignore here.

Reply:
Yeah, I don't love it either.. it's not great.

Reply:
You can around this by adding an assert `isinstance` here.